lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -3832,6 +3832,12 @@ Error glow::optimizeFunction(Function *F, const Backend &B,
   }
 
   RETURN_IF_ERR(optimizeFunctionBeforeLowering(F, cctx));
+
+  // Fold before lowering to enable cases which would not fold after
+  // lowering, particularly convolution+relu since relu will be lowered
+  // to max(0, x).
+  ::glow::fold(F, cctx, &B);
```

Comment:
Isn't `glow::fold` called from `optimizeFunctionBeforeLowering`, which is invoked in line 3834? The only difference I see is that you pass the backend `B` as a parameter now. So, maybe `optimizeFunctionBeforeLowering` should take a backend as a parameter and pass it down to `fold` later?

Reply:
Ah, so I actually made that change in https://github.com/pytorch/glow/pull/3867 but I guess it got lost in the holiday shuffle and never landed 🙁 

Thinking more about it though, I don't think we actually can fuse until after `transformForPrecisionMode()` -- otherwise the graph may look differently and so the quantization profile may no longer fit the graph.

I think the right thing to do here is actually move the call to `foldActivations()` out of `fold()` to right after the call to `transformForPrecisionMode()`

Reply:
Done. To avoid the relu to max lowering that hinders convolution+relu conversions happening later, I forbid relu to max lowering in the OpenCL backend and just implement RelU in the backend (using max). This should be beneficial also due to the constant scalar 0 input to the max (no need for splatted matrix just for this).